api/pkg/apis/v1alpha1/providers/target/k8s/k8s.go

```diff
@@ -562,6 +652,21 @@ func (i *K8sTargetProvider) upsertDeployment(ctx context.Context, namespace stri
 	if err != nil {
 		return err
 	}
+
+	waitErr := i.pollCheck(ctx, false, func() error {
+		_, err := i.Client.AppsV1().Deployments(namespace).Get(ctx, name, metav1.GetOptions{})
```

Comment:
So sorry I am just aware you only poll the deployment object. This is not enough. We need to do sth like "kubectl rollout status deployment/<deploymentName>"

https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#complete-deployment

sample code will be,
```go
// Check rollout status
    err = wait.PollImmediate(time.Second, time.Minute*10, func() (bool, error) {
        deployment, err = clientset.AppsV1().Deployments(namespace).Get(context.TODO(), deploymentName, metav1.GetOptions{})
        if err != nil {
            return false, err
        }

        for _, condition := range deployment.Status.Conditions {
            if condition.Type == appsv1.DeploymentProgressing && condition.Status == "True" && condition.Reason == "NewReplicaSetAvailable" {
                fmt.Println("Deployment successfully rolled out")
                return true, nil
            }
        }

        return false, nil
    })
```

Reply:
fixed. follow the same check pattern with helm